src/lock/postgres.rs

```diff
@@ -0,0 +1,182 @@
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use std::time::{Duration, SystemTime};
+
+use anyhow::{Context, Result};
+use tokio::sync::Mutex;
+use tokio_postgres::{Client, NoTls};
+use tracing::{debug, warn};
+
+use super::DistributedLock;
+
+/// PostgreSQL-based distributed lock using advisory locks.
+///
+/// This lock uses PostgreSQL advisory locks to coordinate access across multiple hosts.
+/// The lock is identified by a 64-bit integer derived from hashing the target path.
+pub struct PostgresLock {
+    client: Arc<Mutex<Client>>,
+    lock_key: i64,
+    target_path: PathBuf,
+    owner: Option<String>,
+    acquired_at: SystemTime,
+}
+
+impl PostgresLock {
+    /// Create a new PostgreSQL lock client.
+    ///
+    /// # Arguments
+    /// * `connection_string` - PostgreSQL connection string (e.g., "host=localhost user=postgres")
+    pub async fn connect(connection_string: &str) -> Result<Arc<Mutex<Client>>> {
+        let (client, connection) = tokio_postgres::connect(connection_string, NoTls)
+            .await
+            .context("failed to connect to PostgreSQL")?;
+
+        // Spawn connection handler
+        tokio::spawn(async move {
+            if let Err(e) = connection.await {
+                warn!(error = %e, "PostgreSQL connection error");
+            }
+        });
+
+        Ok(Arc::new(Mutex::new(client)))
+    }
+
+    /// Convert a path to a lock key by hashing it to a 64-bit integer.
+    fn path_to_lock_key(path: &Path) -> i64 {
+        let mut hasher = DefaultHasher::new();
+        path.hash(&mut hasher);
+        hasher.finish() as i64
+    }
+}
+
+impl DistributedLock for PostgresLock {
+    async fn try_acquire(
+        target: &Path,
+        _ttl: Duration,
+        owner: Option<&str>,
+    ) -> Result<Option<Self>> {
+        // Get PostgreSQL connection string from environment
+        let connection_string = std::env::var("SWANLAKE_LOCK_POSTGRES_CONNECTION")
```

Comment:
let's delete this block, and also SWANLAKE_LOCK_POSTGRES_CONNECTION

we only use the default env string, which should be enough.